### DISCUSSION TITLE
n-drawer 抽屉为什么要设计一个 n-drawer-content 的组件在内部？

### DISCUSSION THREAD

**Discussion by ManiaJack** [tusen-ai/naive-ui#3419] [General]
似乎 content 组件提供 title/footer 两个 slot，提供了 title/closeable 等不多的 props，为什么不把这些内容直接放在 n-drawer 组件上呢？

我目前没有想到什么情况是只需要 n-drawer 不需要 n-drawer-content 的，示例代码也全都有 content，感觉在用的时候多一层组件既麻烦又显得不那么有意义。

**Comment by 07akioni** [comment]
因为自定义滚动条把这个事情搞得复杂了

**Reply by 07akioni** [reply]
自定义滚动条和原生滚动条的行为是不一致的，从原理上也不可能一致，所以需要搞个东西在两边都能 work 起来
